engine/consensus/dkg/messaging_engine.go

```diff
@@ -109,10 +119,20 @@ func (e *MessagingEngine) forwardOutgoingMessages() {
 	for {
 		select {
 		case msg := <-e.tunnel.MsgChOut:
-			err := e.conduit.Unicast(&msg.DKGMessage, msg.DestID)
+			expRetry, err := retry.NewExponential(RETRY_MILLISECONDS)
 			if err != nil {
-				e.log.Err(err).Msg("error sending dkg message")
+				e.log.Fatal().Err(err).Msg("failed to create retry mechanism")
 			}
+
+			maxedExpRetry := retry.WithMaxRetries(RETRY_MAX, expRetry)
+			err = retry.Do(e.unit.Ctx(), maxedExpRetry, func(ctx context.Context) error {
```

Comment:
Is this OK that we block for the retries here? Both answers are valid, I don't have enough knowledge, just wanted to make sure we are OK with blocking until the message is propagated.
This will also block eventual quitting.

Reply:
Yes this is okay we want to make sure message is propagated to avoid unnecessary white board messages during failure.